WalletWasabi/Tor/Socks5/Pool/TorHttpPool.cs

```diff
@@ -157,17 +157,27 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, ICi
 				TorTcpConnection? connectionToDispose = null;
 				OneOffCircuit? oneOffCircuitToDispose = null;
 
+				INamedCircuit? namedCircuit = null;
+
+				if (circuit is AnyOneOffCircuit)
+				{
+					oneOffCircuitToDispose = new();
+					namedCircuit = oneOffCircuitToDispose;
+				}
+				else if (circuit is INamedCircuit withName)
+				{
+					namedCircuit = withName;
+				}
+				else
+				{
+					throw new NotImplementedException("This should never happen.");
+				}
+
 				try
 				{
-					connection = await ObtainFreeConnectionAsync(request, circuit, cancellationToken).ConfigureAwait(false);
+					connection = await ObtainFreeConnectionAsync(request, namedCircuit, cancellationToken).ConfigureAwait(false);
 					connectionToDispose = connection;
 
-					// AnyOneOffCircuit is a special choice and it must lead to OneOffCircuit disposal every time. 
-					if (circuit is AnyOneOffCircuit && connection.Circuit is OneOffCircuit oneOffCircuit)
-					{
-						oneOffCircuitToDispose = oneOffCircuit;
-					}
-
 					Logger.LogTrace($"['{connection}'][Attempt #{i}] About to send request.");
 					HttpResponseMessage response = await SendCoreAsync(connection, request, cancellationToken).ConfigureAwait(false);
```

Comment:
In case of SendCoreAsync fail, where are we increasing the IsolationId?

Reply:
I made a test. Set RegisterInputAsync retrytimeout to 1 sec here: 

https://github.com/molnard/WalletWasabi/blob/master/WalletWasabi/WabiSabi/Client/WabiSabiHttpApiClient.cs#L39

It will retry on the same circuit again and again - not increasing the IsolationId.